pkg/sql/sqlbase/table.go

```diff
@@ -1525,30 +1526,74 @@ func (a byID) Len() int           { return len(a) }
 func (a byID) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
 func (a byID) Less(i, j int) bool { return a[i].id < a[j].id }
 
+func EncodeInvertedIndexKeys(tableDesc *TableDescriptor, index *IndexDescriptor, colMap map[ColumnID]int,
+	values []tree.Datum, keyPrefix []byte) (key [][]byte, err error) {
+	if len(index.ColumnIDs) > 1 {
+		return nil, pgerror.NewError(pgerror.CodeInternalError,
+			"trying to apply inverted index to more than one column")
+	}
+
+	var val tree.Datum
+	if i, ok := colMap[index.ColumnIDs[0]]; ok {
+		val = values[i]
+	} else {
+		val = tree.DNull
+	}
+
+	return EncodeInvertedIndexTableKeys(val, keyPrefix)
+}
+
+func EncodeInvertedIndexTableKeys(val tree.Datum, inKey []byte) (key [][]byte, err error) {
```

Comment:
These functions need some comments.. The naming is confusing, `EncodeInvertedIndexKeys` calls `EncodeInvertedIndexTableKeys` which calls `JSON.EncodeInvertedIndexKeys`..

In particular, it should be mentioned what this API does with `inKey` (based on the implementation, every resulting key has a copy of `inKey` as a prefix?)